sdk/tests/conformance/textures/misc/texture-video-transparent.html

```diff
@@ -0,0 +1,166 @@
+** permit persons to whom the Materials are furnished to do so, subject to
+** the following conditions:
+**
+** The above copyright notice and this permission notice shall be included
+** in all copies or substantial portions of the Materials.
+**
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>Upload texture from animating transparent WebM</title>
+<link rel="stylesheet" href="../../../resources/js-test-style.css"/>
+<script src="../../../js/js-test-pre.js"></script>
+<script src="../../../js/webgl-test-utils.js"></script>
+<script>
+"use strict";
+const wtu = WebGLTestUtils;
+let gl;
+let textureLoc;
+let successfullyParsed;
+let video;
+
+initTestingHarness();
+
+function init()
+{
+    description("Upload texture from animating transparent WebM");
+
+    const canvas = document.getElementById("example");
+    gl = wtu.create3DContext(canvas);
+    const program = wtu.setupTexturedQuad(gl);
+
+    textureLoc = gl.getUniformLocation(program, "tex");
+
+    video = document.getElementById("vid");
+    const type = 'video/webm; codecs="vp8"';
+    if (!video.canPlayType) {
+        testFailed("video.canPlayType required method missing");
+        finishTest();
+        return;
+    }
+    if (!video.canPlayType(type).replace(/no/, '')) {
+        debug(info.type + " unsupported");
+        finishTest();
+        return;
+    };
+    wtu.startPlayingAndWaitForVideo(video, runTest);
+}
+
+function runTest(videoElement)
+{
+    let i = 0;
+    requestAnimationFrame(function frame() {
+        runOneIteration(videoElement, false);
+        runOneIteration(videoElement, true);
+
+        ++i;
+        if (i < 120) {
+            requestAnimationFrame(frame);
+        } else {
+            wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors");
+            finishTest();
+        }
+    });
+
+}
+
+function runOneIteration(videoElement, useTexSubImage2D)
+{
+    const texture = gl.createTexture();
+    // Bind the texture to texture unit 0
+    gl.bindTexture(gl.TEXTURE_2D, texture);
+    // Upload the videoElement into the texture
+    if (useTexSubImage2D) {
+        // Initialize the texture to black first
+        gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA,
+                      videoElement.videoWidth, videoElement.videoHeight, 0,
+                      gl.RGBA, gl.UNSIGNED_BYTE, null);
+        gl.texSubImage2D(gl.TEXTURE_2D, 0, 0, 0, gl.RGBA, gl.UNSIGNED_BYTE, videoElement);
```

Comment:
Also, will this video always have data after body.onload? There's no chance it's still waiting on data for the first frame? Can we assert that we haveData? (or whatever that attribute is? I think Ken knows)

Reply:
This runs after wtu.startPlayingAndWaitForVideo above.

Reply:
Right, startPlayingAndWaitForVideo should handle this.